### PR TITLE
feat: [PL-57734]: Increased Helm Unit Tests coverage

### PR DESCRIPTION
**Pull request by GokulBansal0**
Updated unit test for helm charts

### CODE REVIEW

File: tests/deployment_test.yaml [hunk 1]
```diff
@@ -80,54 +82,77 @@ tests:
...
-  #   templates:
-  #     - deployment.yaml
-  #   set:
-  #     autoscaling.enabled: false
-  #     replicaCount: 3
-  #   asserts:
-  #     - equal:
-  #         path: spec.replicas
-  #         value: 3
+  - it: validate replicaCount when autoscaling is enabled
```
Comment (avaakash) [inline tests/deployment_test.yaml:120]: This might be an issue for deployments that do not support HPA.

File: tests/deployment_test.yaml [hunk 2]
```diff
@@ -316,8 +393,8 @@ tests:
           operator: "Equal"
           value: "value"
           effect: "NoSchedule"
-      # maxSurge: 4
-      # maxUnavailable: 2
-      # replicaCount: 3
+#       maxSurge: 4
+#       maxUnavailable: 2
+#       replicaCount: 3
```
Comment (avaakash) [inline tests/deployment_test.yaml:398]: remove if not required

File: tests/hpa_test.yaml [hunk 3]
```diff
@@ -1,16 +1,21 @@
 ---
 suite: test horizontal pod autoscaler
-
 templates:
   - hpa.yaml
-
 set:
   autoscaling.enabled: true
-
+  nameOverride: my-chart-name
```
Comment (avaakash) [inline tests/hpa_test.yaml:7]: Set this inside the test, we need to only validate this is working, other test should use the default value

File: tests/hpa_test.yaml [hunk 4]
```diff
@@ -56,18 +61,28 @@ tests:
...
           path: spec.scaleTargetRef
-      # - equal:
-      #     path: spec.scaleTargetRef.kind
-      #     value: Deployment
-      # - equal:
-      #     path: spec.scaleTargetRef.name
-      #     value: metadata.name
+      - equal:
+           path: spec.scaleTargetRef.kind
+           value: Deployment
```
Comment (avaakash) [inline tests/hpa_test.yaml:81]: HPA can be for Statefulset as well

Comment (avaakash) [reply tests/hpa_test.yaml:81]: Need to research on how we can make these test cases dynamic

File: tests/statefulset_test.yaml [hunk 5]
```diff
@@ -80,24 +82,53 @@ tests:
...
+       - statefulset.yaml
+    set:
+      autoscaling.enabled: false
+      replicaCount: 3
+    asserts:
+      - equal:
+         path: spec.replicas
+         value: 3
+
+  - it: validate replicaCount when autoscaling is disabled
```
Comment (avaakash) [inline tests/statefulset_test.yaml:96]: Same comments as deployment

File: tests/hpa_test.yaml (hunk 3 above)
Comment (GokulBansal0) [reply tests/hpa_test.yaml]: okay

**Comment by avaakash** [comment]
@GokulBansal0 please update the PR description

**Review by kapilgarg1996 — APPROVED** [review]

**Review by avaakash — APPROVED** [review]

**Pull request merged**